sqlglot/dialects/snowflake.py

```diff
@@ -682,14 +682,23 @@ def _parse_location_property(self) -> exp.LocationProperty:
             return self.expression(exp.LocationProperty, this=self._parse_location_path())
 
         def _parse_file_location(self) -> t.Optional[exp.Expression]:
-            return self._parse_table_parts()
+            # Parse either a subquery or a staged file
+            return (
+                self._parse_select(table=True)
+                if self._match(TokenType.L_PAREN, advance=False)
+                else self._parse_table_parts()
+            )
```

Comment:
Can't we just do `self._parse_table()` here?

Reply:
This would indeed combine the two but it will consume the next token as table alias, right?

Reply:
Ah I see, yeah makes sense